sw/host/opentitanlib/src/transport/hyperdebug/uart.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+
+use std::cell::RefCell;
+use std::time::Duration;
+use std::io::Read;
+use std::io::Write;
+use std::path::Path;
+
+use crate::io::uart::Uart;
+use crate::transport::hyperdebug::Error;
+use crate::transport::hyperdebug::Hyperdebug;
+
+pub struct HyperdebugUart {
+    port: RefCell<Box<dyn serialport::SerialPort>>,
+}
+
+impl HyperdebugUart {
+    pub fn open(_hyperdebug: &Hyperdebug, tty: &Path) -> Result<Self> {
+        let port =
+            serialport::new(tty.to_str().ok_or(Error::UnicodePathError)?,
+                            115_200)
+            .timeout(Duration::from_millis(100))
+            .open().expect("Failed to open port");
+        Ok(HyperdebugUart {
+            port: RefCell::new(port),
+        })
+    }
+}
+
+impl Uart for HyperdebugUart {
+
+    fn read(&self, buf: &mut [u8]) -> Result<usize> {
+        self.port.borrow_mut().set_timeout(Duration::MAX)?;
```

Comment:
I had trouble with `Duration::MAX` and serialport library:
https://github.com/lowRISC/opentitan/blob/master/sw/host/opentitanlib/src/transport/cw310/uart.rs#L20

Reply:
Thanks for the warning.